main.tf

```
@@ -56,5 +56,16 @@ resource "google_cloudfunctions_function" "function" {
     }
   }
 
+  dynamic "secret_environment_variables" {
+    for_each = var.secret_environment_variables != null ? [true] : []
+
+    content {
+      key        = var.secret_environment_variables.key
+      project_id = var.secret_environment_variables.project_id
```

project_id is optional 

```suggestion
      project_id = try(var.secret_environment_variables.project_id, null)
```

---

Of course, good catch!